### PR TITLE
deepin.deepin-image-viewer: fix build with libraw 0.21.1

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes

Fetch a patch from Arch Linux

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wineee** [comment]
@ofborg build deepin.deepin-image-viewer

File: pkgs/desktops/deepin/apps/deepin-image-viewer/default.nix [hunk 1]
```diff
@@ -35,6 +35,11 @@ stdenv.mkDerivation rec {
       url = "https://github.com/linuxdeepin/deepin-image-viewer/commit/4a046e6207fea306e592fddc33c1285cf719a63d.patch";
       sha256 = "sha256-aIgYmq6WDfCE+ZcD0GshxM+QmBWZGjh9MzZcTMrhBJ0=";
     })
+    (fetchpatch {
+      name = "fix build with libraw 0.21";
+      url = "https://raw.githubusercontent.com/archlinux/svntogit-community/packages/deepin-image-viewer/trunk/libraw-0.21.patch";
```
Comment (NickCao) [inline pkgs/desktops/deepin/apps/deepin-image-viewer/default.nix:40]: This url is not permanent, please point it to a specific commit.

Comment (wineee) [reply pkgs/desktops/deepin/apps/deepin-image-viewer/default.nix]: Commit has PKGBUILD changes，Changed to directly submit this patch file

**Comment by wineee** [comment]
@ofborg build deepin.deepin-image-viewer

File: pkgs/desktops/deepin/apps/deepin-image-viewer/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/desktops/deepin/apps/deepin-image-viewer/default.nix]: https://raw.githubusercontent.com/archlinux/svntogit-community/2ff11979704dd7156a7e7c3bae9b30f08894063d/trunk/libraw-0.21.patch

**Comment by NickCao** [comment]
@ofborg build deepin.deepin-image-viewer

**Review by NickCao — APPROVED** [review]

**Pull request merged**